index.js

```diff
@@ -0,0 +1,208 @@
+		// The store is reset eveytime the applet is reload.
+		this.store.clear();
+	}
+
+	async getPrice() {
+		// Get the current price of the selected flight. 
+		logger.info(`Getting price`);
+		const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		const API_WORLD_URL = `browsequotes/v1.0/US/${this.config.currency}/en-US`;
+		const place = `${this.config.originPlace}-sky/${this.config.destinationPlace}-sky`;
+		const date = `${this.config.departureDate}?inboundpartialdate=${this.config.returnDate}`;
+		const settings = {
+			url: `${API_BASE_URL}/${API_WORLD_URL}/${place}/${date}`,
+			method: 'GET',
+			headers: {
+				'x-rapidapi-host': 'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
+				'x-rapidapi-key': this.authorization.apiKey
+			}
+		}
+		if (!this.authorization.apiKey) {
+			throw 'Missing API key';
+		}
+		return request(settings).then(answer => {
+			const collectPrice = JSON.parse(answer);
+			// If no prices are listed
+			// It will display `This flight is not listed. Please modify your request.` 
+			// See the logic line 118
+			if (collectPrice.Quotes.length == 0) {
+				return null;
+			}
+			// Collects the first element of "Quotes" corresponding 
+			// to the minprice for the selected flight.
+			return collectPrice.Quotes[0].MinPrice;
+		});
+	}
+
+	// Search method linked to a JSON file for the airport name input in the form.
+	// The JSON file holds the IATA code and the name of the airports
+	async options(fieldId, search) {
+		// const API_BASE_URL = `https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices`;
+		if (airports) {
+			logger.info("Sending preloaded airports");
+			return this.getIATA(airports, search);
+		} else if (fs.existsSync('./airports.json')) {
+			logger.info('Loading places from a file.');
+			const airports = require('./airports.json');
+			return this.getIATA(airports, search);
+		}
+	}
+
+	/**
+	* @param {*} airports
+	* @param {String} search 
+	*/
+	async getIATA(airports, search) {
+		if (search != null) {
+			search = search.trim().toLowerCase();
+		}
+		let option = [];
+		for (const airport of airports) {
+			const key = airport.iata;
+			let value = airport.name || '';
+			if (!search || value.toLowerCase().includes(search)) {
+				option.push({
+					key: key,
+					value: value
+				});
+			}	
+		}
+		return option;
+	}	
+
+	// Store price obtained from first loaded.
+	// The price is updated every 24 hours.
+	// This price is compared with the price collected from the API every minute.
+	setFirstPrice(price) {
+		const firstPriceDate = this.store.get("firstPriceDate");
+		if(firstPriceDate == null || new Date().getTime() - firstPriceDate > 24*3600*1000) {
+			this.store.put("firstPrice", price);
+			this.store.put("firstPriceDate", new Date().getTime());
+		}
+		return true;
+	}
+
+	// Retrieve stored price data.
+	getFirstPrice() {
+		return this.store.get("firstPrice");
+	}
```

Comment:
I left comments on the previews pull request.. You didn't resolve it... Refers to the previous pull request to see the comments